src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -74,319 +74,298 @@ const defaultProps = {
-            draftMessage = Str.htmlDecode(this.props.draftMessage);
+            return parser.htmlToMarkdown(props.draftMessage).trim();
         }
-
-        this.state = {
-            draft: draftMessage,
-            selection: {
-                start: 0,
-                end: 0,
-            },
-            isFocused: false,
-            hasExceededMaxCommentLength: false,
-        };
-    }
-
-    componentDidMount() {
+        // We need to decode saved draft message because it's escaped before saving.
+        return Str.htmlDecode(props.draftMessage);
+    });
+    const [selection, setSelection] = useState({start: 0, end: 0});
+    const [isFocused, setIsFocused] = useState(false);
+    const [hasExceededMaxCommentLength, setHasExceededMaxCommentLength] = useState(false);
+
+    const textInputRef = useRef(null);
+    const isFocusedRef = useRef(false);
+
+    useEffect(() => {
+        // required for keeping last state of isFocused variable
+        isFocusedRef.current = isFocused;
+    }, [isFocused]);
+
+    useEffect(() => {
         // For mobile Safari, updating the selection prop on an unfocused input will cause it to automatically gain focus
         // and subsequent programmatic focus shifts (e.g., modal focus trap) to show the blue frame (:focus-visible style),
         // so we need to ensure that it is only updated after focus.
-        this.setState((prevState) => ({
-            selection: {
-                start: prevState.draft.length,
-                end: prevState.draft.length,
-            },
-        }));
-    }
-
-    componentWillUnmount() {
-        // Skip if this is not the focused message so the other edit composer stays focused.
-        if (!this.state.isFocused) {
-            return;
-        }
+        setDraft((prevDraft) => {
+            setSelection({
+                start: prevDraft.length,
+                end: prevDraft.length,
+            });
+            return prevDraft;
+        });
 
-        // Show the main composer when the focused message is deleted from another client
-        // to prevent the main composer stays hidden until we swtich to another chat.
-        ComposerActions.setShouldShowComposeInput(true);
-    }
+        return () => {
+            // Skip if this is not the focused message so the other edit composer stays focused
+            if (!isFocusedRef.current) {
+                return;
+            }
 
-    /**
-     * Update Selection on change cursor position.
-     *
-     * @param {Event} e
-     */
-    onSelectionChange(e) {
-        this.setState({selection: e.nativeEvent.selection});
-    }
+            // Show the main composer when the focused message is deleted from another client
+            // to prevent the main composer stays hidden until we swtich to another chat.
+            ComposerActions.setShouldShowComposeInput(true);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- we need it to call it once on mount
```

Comment:
```suggestion
```

Not needed, we are using a ref now and it always have the same value so no need for it to be a dependency (and no error will be thrown).